src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -1012,8 +1011,9 @@ case class MatrixAggregateColsByKey(child: MatrixIR, aggIR: IR) extends MatrixIR
         rvb.start(newRVType)
         rvb.startStruct()
         var k = 0
-        while (k < newEntriesIndex) {
-          rvb.addField(oldRVRowType, rv, k)
+        while (k < newRVType.size) {
+          if (k != oldEntriesIndex)
```

Comment:
I wrote this code initially -- where do we guarantee that the last field in the new region value is the array of entries?

Reply:
in the matrix type, we use `child.typ.copyParts(entryType=...)`. This moves the entries to the end.

The other option would be to change the matrix type instead.